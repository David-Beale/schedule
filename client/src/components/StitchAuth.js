import React, {
  useEffect,
  createContext,
  useContext,
  useState,
  useMemo
} from 'react';

import {
  hasLoggedInUser,
  loginAnonymous,
  loginGoogle,
  logoutCurrentUser,
  getCurrentUser,
  addAuthenticationListener,
  removeAuthenticationListener,
  handleOAuthRedirects
} from './../stitch/authentication';

const StitchAuthContext = createContext();

export function useStitchAuth() {
  const context = useContext(StitchAuthContext);
  if (!context) {
    throw new Error(`useStitchAuth must be used within a StitchAuthProvider`);
  }
  return context;
}

export function StitchAuthProvider(props) {
  const [authState, setAuthState] = useState({
    isLoggedIn: hasLoggedInUser(),
    currentUser: getCurrentUser()
  });

  useEffect(() => {
    const authListener = {
      onUserLoggedIn: (auth, loggedInUser) => {
        if (loggedInUser) {
          setAuthState((authState) => ({
            ...authState,
            isLoggedIn: true,
            currentUser: loggedInUser
          }));
        }
      },
      // eslint-disable-next-line no-unused-vars
      onUserLoggedOut: (auth, loggedOutUser) => {
        setAuthState((authState) => ({
          ...authState,
          isLoggedIn: false,
          currentUser: null
        }));
      }
    };
    addAuthenticationListener(authListener);
    handleOAuthRedirects();
    setAuthState((state) => ({ ...state }));
    return () => {
      removeAuthenticationListener(authListener);
    };
  }, []);

  // Authentication Actions
  const handleLogin = async (provider) => {
    if (!authState.isLoggedIn) {
      switch (provider) {
        case 'anonymous':
          return loginAnonymous();
        case 'google':
          console.log('hoohle!');
          return loginGoogle();
        // eslint-disable-next-line no-empty
        default: {
        }
      }
    }
  };
  const handleLogout = async () => {
    const { isLoggedIn } = authState;
    if (isLoggedIn) {
      await logoutCurrentUser();
      setAuthState({
        ...authState,
        isLoggedIn: false,
        currentUser: null
      });
    } else {
      console.log(`can't handleLogout when no user is logged in`);
    }
  };

  // We useMemo to improve performance by eliminating some re-renders
  const authInfo = useMemo(() => {
    const { isLoggedIn, currentUser } = authState;
    const value = {
      isLoggedIn,
      currentUser,
      actions: { handleLogin, handleLogout }
    };
    return value;
  }, [authState.isLoggedIn]);
  return (
    <StitchAuthContext.Provider value={authInfo}>
      {props.children}
    </StitchAuthContext.Provider>
  );
}
