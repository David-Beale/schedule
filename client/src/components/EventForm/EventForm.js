import React, { useState } from 'react';
import { Form, Input, DatePicker, Button, TimePicker, Upload } from 'antd';
import { LoadingOutlined, PlusOutlined } from '@ant-design/icons';
import { addEvent } from '../../redux/actions';
import { useStitchAuth } from '../StitchAuth/StitchAuth';
import { useDispatch } from 'react-redux';
import { Route, Redirect } from 'react-router-dom';
const { TextArea } = Input;
const { RangePicker } = TimePicker;

function EventForm() {
  const dispatch = useDispatch();
  const { currentUser } = useStitchAuth();
  const [loading, setLoading] = useState(false);
  const [imageUrl, setImageUrl] = useState('');
  const [submitted, setSubmitted] = useState(false);

  const layout = {
    labelCol: { span: 8 },
    wrapperCol: { span: 14 }
  };

  async function submitForm(values) {
    const {
      title,
      description,
      eventTime,
      eventDate,
      artistName,
      streamUrl
    } = values;
    const formData = {
      artistName,
      title,
      description,
      streamUrl,
      image: imageUrl,
      date: new Date(
        eventDate.format('MMMM D, YYYY ') + eventTime[0].format('HH:mm') + ':00'
      ),
      endTime: new Date(
        eventDate.format('MMMM D, YYYY ') + eventTime[1].format('HH:mm') + ':00'
      )
    };

    dispatch(addEvent(formData, currentUser.id));
    setSubmitted(true);
  }

  function uploadPicture(info) {
    if (info.file.status === 'uploading') {
      setLoading(true);
      return;
    }
  }

  const uploadButton = (
    <div>
      {loading ? <LoadingOutlined /> : <PlusOutlined />}
      <div className="ant-upload-text">Upload</div>
    </div>
  );

  const cloudinaryRequest = (e) => {
    const formData = new FormData();
    formData.append('upload_preset', 'eumh276f');
    formData.append('file', e.file);
    fetch(
      `https://api.cloudinary.com/v1_1/${process.env.REACT_APP_CLOUDNAME}/image/upload`,
      {
        method: 'POST',
        body: formData
      }
    )
      .then((response) => response.json())
      .then((data) => {
        setImageUrl(data.url);
      })
      .catch((error) => {
        console.error('Error:', error);
      });
  };

  return (
    <Route>
      <div
        style={{
          display: 'flex',
          justifyContent: 'center',
          flexDirection: 'column'
        }}
      >
        <h1 style={{ margin: '50px auto auto auto', paddingBottom: '3vh' }}>
          Create a new event
        </h1>
        <Form
          labelCol={{ span: 8 }}
          wrapperCol={{ span: 14 }}
          {...layout}
          style={{ width: '60vw', height: '50vh' }}
          onFinish={submitForm}
        >
          <Form.Item
            name="artistName"
            label="Artist's Name"
            rules={[
              { required: true, message: "Please write the artist's name" }
            ]}
          >
            <Input></Input>
          </Form.Item>
          <Form.Item
            name="title"
            label="Title"
            rules={[
              { required: true, message: "Please write the event's title" }
            ]}
          >
            <Input></Input>
          </Form.Item>
          <Form.Item
            name="description"
            label="Description"
            rules={[
              {
                required: true,
                message: 'Please write a description for the event'
              },
              {
                min: 50,
                message: 'Description must be at least 50 characters long'
              }
            ]}
          >
            <TextArea autoSize={{ minRows: 4, maxRows: 7 }} />
          </Form.Item>
          <Form.Item
            name="upload"
            label="Profile picture"
            rules={[
              {
                required: true,
                message: 'Please upload a picture for the event'
              }
            ]}
          >
            <Upload
              name="avatar"
              customRequest={cloudinaryRequest}
              accept=".png, .jpg, .jpeg"
              listType="picture-card"
              className="avatar-uploader"
              showUploadList={false}
              action="https://www.mocky.io/v2/5cc8019d300000980a055e76"
              onChange={uploadPicture}
            >
              {imageUrl ? (
                <img src={imageUrl} alt="avatar" style={{ width: '100%' }} />
              ) : (
                uploadButton
              )}
            </Upload>
          </Form.Item>
          <Form.Item
            name="streamUrl"
            label="Stream URL"
            rules={[
              { required: true, message: 'Please input a stream URL' },
              {
                type: 'url',
                message: 'Please input a valid stream URL.'
              }
            ]}
          >
            <Input></Input>
          </Form.Item>

          <Form.Item
            name="patreonUrl"
            label="Patreon URL"
            rules={[
              { required: true, message: 'Please input a valid Patreon URL' },
              {
                type: 'url',
                essage: 'Please input a valid Patreon URL'
              }
            ]}
          >
            <Input></Input>
          </Form.Item>

          <Form.Item
            name="eventDate"
            label="Date"
            rules={[
              { required: true, message: 'Please choose a date for the event' }
            ]}
          >
            <DatePicker />
          </Form.Item>
          <Form.Item
            name="eventTime"
            label="Time"
            rules={[
              {
                required: true,
                message: 'Please choose a time range for the event'
              }
            ]}
          >
            <RangePicker format="HH:mm" />
          </Form.Item>
          <Form.Item wrapperCol={{ ...layout.wrapperCol, offset: 8 }}>
            <Button type="primary" htmlType="submit">
              Submit
            </Button>
          </Form.Item>
        </Form>
      </div>
    </Route>
  );
}

export { EventForm as default };
