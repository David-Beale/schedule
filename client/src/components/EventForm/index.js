import EventForm from './EventForm';

export { EventForm as default };
