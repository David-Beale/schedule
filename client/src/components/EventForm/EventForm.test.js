import EventForm from './EventForm';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from '../../redux/store';
import { StitchAuthProvider } from '../StitchAuth/StitchAuth';
import { BrowserRouter as Router } from 'react-router-dom';

const store = configureStore();

describe('EventForm', () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <Router>
          <StitchAuthProvider>
            <EventForm />
          </StitchAuthProvider>
        </Router>
      </Provider>,
      div
    );
  });
  it('should render properly', () => {
    const component = render(
      <Provider store={store}>
        <Router>
          <StitchAuthProvider>
            <EventForm />
          </StitchAuthProvider>
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
