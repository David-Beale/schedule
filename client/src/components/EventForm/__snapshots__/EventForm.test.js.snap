// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <textarea
      aria-hidden="true"
      style="letter-spacing:normal;line-height:normal;padding-top:2px;padding-bottom:2px;font-family:-webkit-small-control;font-weight:;font-size:;font-variant:;text-rendering:auto;text-transform:none;width:;text-indent:0;padding-left:2px;padding-right:2px;border-width:1px;box-sizing:;
  min-height:0 !important;
  max-height:none !important;
  height:0 !important;
  visibility:hidden !important;
  overflow:hidden !important;
  position:absolute !important;
  z-index:-1000 !important;
  top:0 !important;
  right:0 !important
"
      tab-index="-1"
    />
    <div>
      <div
        style="display: flex; justify-content: center; flex-direction: column;"
      >
        <h1
          style="margin: 50px auto auto auto; padding-bottom: 3vh;"
        >
          Create a new event
        </h1>
        <form
          class="ant-form ant-form-horizontal"
          style="width: 60vw; height: 50vh;"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-8 ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="artistName"
                title="Artist's Name"
              >
                Artist's Name
              </label>
            </div>
            <div
              class="ant-col ant-col-14 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="artistName"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-8 ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="title"
                title="Title"
              >
                Title
              </label>
            </div>
            <div
              class="ant-col ant-col-14 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="title"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-8 ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="description"
                title="Description"
              >
                Description
              </label>
            </div>
            <div
              class="ant-col ant-col-14 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <textarea
                    class="ant-input"
                    id="description"
                    style="height: -28px; min-height: -16px; max-height: -28px; overflow-y: hidden; overflow-x: hidden;"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-8 ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="upload"
                title="Profile picture"
              >
                Profile picture
              </label>
            </div>
            <div
              class="ant-col ant-col-14 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="avatar-uploader ant-upload-picture-card-wrapper"
                  >
                    <div
                      class="ant-upload ant-upload-select ant-upload-select-picture-card"
                    >
                      <span
                        class="ant-upload"
                        role="button"
                        tabindex="0"
                      >
                        <input
                          accept=".png, .jpg, .jpeg"
                          id="upload"
                          style="display: none;"
                          type="file"
                        />
                        <div>
                          <span
                            aria-label="plus"
                            class="anticon anticon-plus"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              class=""
                              data-icon="plus"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <defs>
                                <style />
                              </defs>
                              <path
                                d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                              />
                              <path
                                d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                              />
                            </svg>
                          </span>
                          <div
                            class="ant-upload-text"
                          >
                            Upload
                          </div>
                        </div>
                      </span>
                    </div>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-8 ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="streamUrl"
                title="Stream URL"
              >
                Stream URL
              </label>
            </div>
            <div
              class="ant-col ant-col-14 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="streamUrl"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-8 ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="patreonUrl"
                title="Patreon URL"
              >
                Patreon URL
              </label>
            </div>
            <div
              class="ant-col ant-col-14 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="patreonUrl"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-8 ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="eventDate"
                title="Date"
              >
                Date
              </label>
            </div>
            <div
              class="ant-col ant-col-14 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-picker"
                  >
                    <div
                      class="ant-picker-input"
                    >
                      <input
                        placeholder="Select date"
                        readonly=""
                        size="12"
                        title=""
                        value=""
                      />
                      <span
                        class="ant-picker-suffix"
                      >
                        <span
                          aria-label="calendar"
                          class="anticon anticon-calendar"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="calendar"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-8 ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="eventTime"
                title="Time"
              >
                Time
              </label>
            </div>
            <div
              class="ant-col ant-col-14 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-picker ant-picker-range"
                  >
                    <div
                      class="ant-picker-input ant-picker-input-active"
                    >
                      <input
                        placeholder="Start time"
                        readonly=""
                        size="10"
                        value=""
                      />
                    </div>
                    <div
                      class="ant-picker-range-separator"
                    >
                      <span
                        aria-label="to"
                        class="ant-picker-separator"
                      >
                        <span
                          aria-label="swap-right"
                          class="anticon anticon-swap-right"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="swap-right"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="0 0 1024 1024"
                            width="1em"
                          >
                            <path
                              d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                    <div
                      class="ant-picker-input"
                    >
                      <input
                        placeholder="End time"
                        readonly=""
                        size="10"
                        value=""
                      />
                    </div>
                    <div
                      class="ant-picker-active-bar"
                      style="left: 0px; width: 0px; position: absolute;"
                    />
                    <span
                      class="ant-picker-suffix"
                    >
                      <span
                        aria-label="clock-circle"
                        class="anticon anticon-clock-circle"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="clock-circle"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                          />
                          <path
                            d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-col-14 ant-col-offset-8 ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn ant-btn-primary"
                    type="submit"
                  >
                    <span>
                      Submit
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; justify-content: center; flex-direction: column;"
    >
      <h1
        style="margin: 50px auto auto auto; padding-bottom: 3vh;"
      >
        Create a new event
      </h1>
      <form
        class="ant-form ant-form-horizontal"
        style="width: 60vw; height: 50vh;"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="artistName"
              title="Artist's Name"
            >
              Artist's Name
            </label>
          </div>
          <div
            class="ant-col ant-col-14 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="artistName"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="title"
              title="Title"
            >
              Title
            </label>
          </div>
          <div
            class="ant-col ant-col-14 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="title"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="description"
              title="Description"
            >
              Description
            </label>
          </div>
          <div
            class="ant-col ant-col-14 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <textarea
                  class="ant-input"
                  id="description"
                  style="height: -28px; min-height: -16px; max-height: -28px; overflow-y: hidden; overflow-x: hidden;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="upload"
              title="Profile picture"
            >
              Profile picture
            </label>
          </div>
          <div
            class="ant-col ant-col-14 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <span
                  class="avatar-uploader ant-upload-picture-card-wrapper"
                >
                  <div
                    class="ant-upload ant-upload-select ant-upload-select-picture-card"
                  >
                    <span
                      class="ant-upload"
                      role="button"
                      tabindex="0"
                    >
                      <input
                        accept=".png, .jpg, .jpeg"
                        id="upload"
                        style="display: none;"
                        type="file"
                      />
                      <div>
                        <span
                          aria-label="plus"
                          class="anticon anticon-plus"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="plus"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <defs>
                              <style />
                            </defs>
                            <path
                              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                            />
                            <path
                              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                            />
                          </svg>
                        </span>
                        <div
                          class="ant-upload-text"
                        >
                          Upload
                        </div>
                      </div>
                    </span>
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="streamUrl"
              title="Stream URL"
            >
              Stream URL
            </label>
          </div>
          <div
            class="ant-col ant-col-14 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="streamUrl"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="patreonUrl"
              title="Patreon URL"
            >
              Patreon URL
            </label>
          </div>
          <div
            class="ant-col ant-col-14 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="patreonUrl"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="eventDate"
              title="Date"
            >
              Date
            </label>
          </div>
          <div
            class="ant-col ant-col-14 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-picker"
                >
                  <div
                    class="ant-picker-input"
                  >
                    <input
                      placeholder="Select date"
                      readonly=""
                      size="12"
                      title=""
                      value=""
                    />
                    <span
                      class="ant-picker-suffix"
                    >
                      <span
                        aria-label="calendar"
                        class="anticon anticon-calendar"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="calendar"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-8 ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="eventTime"
              title="Time"
            >
              Time
            </label>
          </div>
          <div
            class="ant-col ant-col-14 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-picker ant-picker-range"
                >
                  <div
                    class="ant-picker-input ant-picker-input-active"
                  >
                    <input
                      placeholder="Start time"
                      readonly=""
                      size="10"
                      value=""
                    />
                  </div>
                  <div
                    class="ant-picker-range-separator"
                  >
                    <span
                      aria-label="to"
                      class="ant-picker-separator"
                    >
                      <span
                        aria-label="swap-right"
                        class="anticon anticon-swap-right"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="swap-right"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="0 0 1024 1024"
                          width="1em"
                        >
                          <path
                            d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                  <div
                    class="ant-picker-input"
                  >
                    <input
                      placeholder="End time"
                      readonly=""
                      size="10"
                      value=""
                    />
                  </div>
                  <div
                    class="ant-picker-active-bar"
                    style="left: 0px; width: 0px; position: absolute;"
                  />
                  <span
                    class="ant-picker-suffix"
                  >
                    <span
                      aria-label="clock-circle"
                      class="anticon anticon-clock-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="clock-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-col-14 ant-col-offset-8 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="submit"
                >
                  <span>
                    Submit
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
